internal/store/builder.go

```diff
@@ -205,6 +207,13 @@ func (b *Builder) WithAllowLabels(labels map[string][]string) {
 	}
 }
 
+// WithAllowMetricLabels configures which labels can be returned for each metric
+func (b *Builder) WithAllowMetricLabels(labels map[string][]string) {
+	if len(labels) > 0 {
+		allowMetricLabelsList = labels
```

Comment:
For POC purposes I used a Package global value to be available to the `wrapPodFunc()` without having to pass it down multiple layers. This was I avoided changing the signature of multiple functions.

KSM doesn't use global variables to share config, however, to implement this across all resources we'll need to change the signature of every `create<Resource><Metric>FamilyGenerator` function of each resource and pass the `allowMetricLabelsList`.